internal/controller/messaging/command.go

```diff
@@ -0,0 +1,223 @@
+)
+
+type key string
+
+const (
+	correlationHeaderKey       key = common.CorrelationHeader
+	CommandRequestTopic            = "CommandRequestTopic"
+	CommandResponseTopicPrefix     = "CommandResponseTopicPrefix"
+)
+
+func SubscribeCommands(ctx context.Context, dic *di.Container) errors.EdgeX {
+	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+	messageBusInfo := container.ConfigurationFrom(dic.Get).MessageQueue
+	requestTopic := messageBusInfo.Topics[CommandRequestTopic]
+	responseTopicPrefix := messageBusInfo.Topics[CommandResponseTopicPrefix]
+
+	messages := make(chan types.MessageEnvelope)
+	messageErrors := make(chan error)
+	topics := []types.TopicChannel{
+		{
+			Topic:    requestTopic,
+			Messages: messages,
+		},
+	}
+
+	messageBus := bootstrapContainer.MessagingClientFrom(dic.Get)
+	err := messageBus.Subscribe(topics, messageErrors)
+	if err != nil {
+		return errors.NewCommonEdgeXWrapper(err)
+	}
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				lc.Infof("Exiting waiting for MessageBus '%s' topic messages", requestTopic)
+				return
+			case err = <-messageErrors:
+				lc.Error(err.Error())
+			case msgEnvelope := <-messages:
+				lc.Debugf("Command request received on message queue. Topic: %s, Correlation-id: %s ", requestTopic, msgEnvelope.CorrelationID)
+
+				// expected command request topic scheme: #/<service-name>/<device-name>/<command-name>/<method>
+				topicLevels := strings.Split(msgEnvelope.ReceivedTopic, "/")
+				length := len(topicLevels)
+				if length < 4 {
+					lc.Error("Failed to parse and construct command response topic scheme, expected request topic scheme: '#/<service-name>/<device-name>/<command-name>/<method>'")
+					continue
+				}
+
+				// expected command response topic scheme: #/<service-name>/<device-name>/<command-name>/<method>
+				serviceName := topicLevels[length-4]
+				deviceName := topicLevels[length-3]
+				commandName := topicLevels[length-2]
+				method := topicLevels[length-1]
+				responseTopic := strings.Join([]string{responseTopicPrefix, serviceName, deviceName, commandName, method}, "/")
+
+				switch strings.ToUpper(method) {
+				case "GET":
+					getCommand(ctx, msgEnvelope, responseTopic, deviceName, commandName, dic)
+				case "SET":
+					setCommand(ctx, msgEnvelope, responseTopic, deviceName, commandName, dic)
+				default:
+					lc.Errorf("unknown command method '%s', only 'get' or 'set' is allowed", method)
+					continue
+				}
+
+				lc.Debugf("Command response published on message queue. Topic: %s, Correlation-id: %s ", responseTopic, msgEnvelope.CorrelationID)
+			}
+		}
+	}()
+
+	return nil
+}
+
+func getCommand(ctx context.Context, msgEnvelope types.MessageEnvelope, responseTopic string, deviceName string, commandName string, dic *di.Container) {
+	var responseEnvelope types.MessageEnvelope
+
+	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+	messageBus := bootstrapContainer.MessagingClientFrom(dic.Get)
```

Comment:
Should just pass these as parameters instead of `dic` since call just pulled them from `dic`. Same for other helper function

Reply:
Unfortunately `application.GetCommand` and `application.SetCommand` still need it

Reply:
Right... Ok.